mol_generator.py

```diff
@@ -0,0 +1,180 @@
+"""
+Using the schrodinger Python modules, read a template .mae file to
+generate the coordgen template C++ files.
+"""
+
+import sys
+import os
+import textwrap
+import argparse
+import subprocess
+
+from schrodinger import structure
+
+NAME = 'CoordgenTemplates'
+
+# String that creates a molecule and adds it to a container
+# of sketcherMinimizerMolecule* at index i.
+#
+# C++ programmers: `{` and `}` are escaped by duplication, so
+# `{{` means `{`.
+_MOLECULE = """
+{{
+    auto molecule = new sketcherMinimizerMolecule();
+    std::array<std::tuple<int, float, float>, {atom_total}> atoms = {{{{
+{atoms}
+    }}}};
+    std::array<std::array<int, 3>, {bond_total}> bonds = {{{{
+{bonds}
+    }}}};
+
+    add_atoms(molecule, atoms);
+    add_bonds(molecule, bonds);
+
+    molecules[i] = molecule;
+    ++i;
+}}
+"""
+
+_DOC = """
+///
+// Find the templates for coordinate generation
+//
+// Autogenerated, do not edit.
+//
+//   $SCHRODINGER/run {script_name} {template_file}
+//
+// generated using {template_file} version {git_hash}.
+//
+"""
+
+_HEADER = """
+#pragma once
+
+{doc}
+
+#include <vector>
+
+class sketcherMinimizerMolecule;
+namespace schrodinger {{
+
+std::vector<sketcherMinimizerMolecule*> coordgen_templates();
```

Comment:
I'd include a comment here indicating that the caller owns the pointers in the `std::vector` this returns.